### ISSUE TITLE
[BUG] Milvus metadata not being retreived

### ISSUE THREAD

**Issue by AbdullahGheith** [langchain4j/langchain4j#707]
**Describe the bug**
I save a textsegment with metadata, but when i retreive it again, the metadata is gone.

**To Reproduce**
```
TextSegment ts = TextSegment.from("I like triangles", Metadata.from("test", "testvalue"));
Embedding embedding = embeddingModel.embed(ts).content();
embeddingStore.add(embedding, ts);
Embedding embed = embeddingModel.embed(new TextSegment(query, Metadata.from("test", "testvalue"))).content();
List<EmbeddingMatch<TextSegment>> relevant = embeddingStore.findRelevant(embed, maxResults);
//relevant list contains 1 entry, but metadata is empty.
```
**Expected behavior**
I expect the metadata to be the same as when i've stored it.

**Please complete the following information:**
- LangChain4j version: 0.27.1
- Java version: e.g. 17
- Spring Boot version (if applicable): 3.1.0

**Comment by AbdullahGheith** [comment]
Note: I just tried replacing milvus with chroma, and it works as intended on Chroma.

**Comment by langchain4j** [comment]
Hi @AbdullahGheith thanks for reporting!

This behavior is documented:
```
/**
 * Represents an <a href="https://milvus.io/">Milvus</a> index as an embedding store.
 * Does not support storing {@link dev.langchain4j.data.document.Metadata} yet.
 */
public class MilvusEmbeddingStore implements EmbeddingStore<TextSegment> {
```

Metadata persistence and filtering will be supported in https://github.com/langchain4j/langchain4j/pull/610 which is going to merged today/tomorrow.

**Comment by AbdullahGheith** [comment]
@langchain4j Thanks.

When finding relevant, can i filter by metadata? So only return results with specific metadata? Or do i just filter after receiving the results?

Offtopic; Which of the (docker runnable) vector databases do you like the most?

**Comment by langchain4j** [comment]
@AbdullahGheith I have no opinion on that, but can recommend starting with `InMemoryEmbeddingStore` if you have a PoC and don't want to invest time in setting up a dedicated embedding store. `InMemoryEmbeddingStore` can be persisted to String/file if you need that.

**Comment by langchain4j** [comment]
Fixed in 0.28.0
